vm/eval.go

```diff
@@ -27,155 +27,133 @@ func Eval(r io.Reader, program *Program, target types.Field) (err error) {
 		}
 	}()
 
-	return evalInner(r, program, target, &pc)
+	_, err = evalInner(r, program, target, &pc)
+	return err
 }
 
-func evalInner(r io.Reader, program *Program, target types.Field, pc *int) (err error) {
+func evalInner(r io.Reader, program *Program, target types.Field, pc *int) (bool, error) {
 	var loop int64
 
 	frame := stackFrame{}
 	for ; *pc < len(program.Instructions); *pc++ {
 		inst := program.Instructions[*pc]
 		switch inst.Op {
 		case Read:
+			var err error
 			switch inst.Operand {
 			case Null:
-				break
+				// Do nothing
 			case Boolean:
 				frame.Boolean, err = readBool(r)
-				break
 			case Int:
 				frame.Int, err = readInt(r)
-				break
 			case Long:
 				frame.Long, err = readLong(r)
-				break
 			case UnusedLong:
 				_, err = readLong(r)
-				break
 			case Float:
 				frame.Float, err = readFloat(r)
-				break
 			case Double:
 				frame.Double, err = readDouble(r)
-				break
 			case Bytes:
 				frame.Bytes, err = readBytes(r)
-				break
 			case String:
 				frame.String, err = readString(r)
-				break
 			default:
 				frame.Bytes, err = readFixed(r, inst.Operand-11)
-				break
 			}
-			break
+			if err != nil {
+				return false, err
+			}
 		case Set:
 			switch inst.Operand {
 			case Null:
-				break
+				// Do nothing
 			case Boolean:
 				target.SetBoolean(frame.Boolean)
-				break
 			case Int:
 				target.SetInt(frame.Int)
-				break
 			case Long:
 				target.SetLong(frame.Long)
-				break
 			case Float:
 				target.SetFloat(frame.Float)
-				break
 			case Double:
 				target.SetDouble(frame.Double)
-				break
 			case Bytes:
 				target.SetBytes(frame.Bytes)
-				break
 			case String:
 				target.SetString(frame.String)
-				break
 			}
-			break
 		case SetDefault:
 			target.SetDefault(inst.Operand)
-			break
 		case Enter:
 			*pc += 1
-			if err = evalInner(r, program, target.Get(inst.Operand), pc); err != nil {
-				return err
+			if accepted, err := evalInner(r, program, target.Get(inst.Operand), pc); err != nil {
```

Comment:
What's the meaning of `accepted`?

Reply:
If `evalInner` were deserializing an optional union, its associated memory would be cleared if not accepted, that is the condition that triggers with the instruction `(exit, null)`